### DISCUSSION TITLE
is it possible to set backgroundColor = .clear to a LNPopupCustomBarViewController ?

### DISCUSSION THREAD

**Discussion by RiccardoManoni** [LeoNatan/LNPopupController#511] [General] [answered]
Is it possible to set backgroundColor = .clear to a LNPopupCustomBarViewController?

I've tried like this but down't work:

`class CustomPopupBarVC: LNPopupCustomBarViewController {

    override func viewDidLoad() {
        view.backgroundColor = .clear
    }

}`

**Comment by LeoNatan** [comment, marked as answer]
It should be possible. Use `LNPopupBarAppearance`.

```swift
let appearance = LNPopupBarAppearance()
appearance.configureWithOpaqueBackground()
appearance.backgroundColor = .clear

containingPopupBar.standardAppearance = appearance
```

**Reply by RiccardoManoni** [reply]
I've tried your code in your example setting a transparent background in CustomMapBarViewController, and it works.
There must be something wrong in my project.

If I use

`
let appearance = LNPopupBarAppearance()
appearance.configureWithOpaqueBackground()
appearance.backgroundColor = .systemPink
containingPopupBar!.inheritsAppearanceFromDockingView = false
containingPopupBar!.standardAppearance = appearance
`


I see the background colored of .systemPink, but if I use instead .clear

`
let appearance = LNPopupBarAppearance()
appearance.configureWithOpaqueBackground()
appearance.backgroundColor = .clear
containingPopupBar!.inheritsAppearanceFromDockingView = false
containingPopupBar!.standardAppearance = appearance
`

the background is not transparent.

This is how I open the popupBar:

self.tabBarController?.popupInteractionStyle = .none
self.tabBarController?.popupBar.standardAppearance.marqueeScrollEnabled = true
self.tabBarController?.popupBar.customBarViewController = customPopupBar
self.tabBarController?.presentPopupBar(withContentViewController: popupContentController, openPopup: true, animated: true, completion: {isPresentingPopup = false; popupIsDismissed = false;})

**Reply by LeoNatan** [reply]
Perhaps `viewDidLoad` is not the correct place to put the code, as the view loading might be triggered before there is a `containingPopupBar`. Try moving the appearance code to where you are presenting the popup bar, and see if that works. I might need to add a `didMoveToPopupBar()` method to `LNPopupCustomBarViewController`, which will be the appropriate place to put this code.

Also, I forgot that there is a `configureWithTransparentBackground()` method, which does what you need, without having to set the background and shadow color to `.clear`.

**Reply by LeoNatan** [reply]
You can also override `viewWillAppear()` and place the code there.

**Reply by RiccardoManoni** [reply]
There is no way, I've tried moving the code inside viewWillAppear and also before the opening of the popup but the background is not transparent.

It seems like there is a container view that contains the customPopupBar and I cannot get a reference of that container.
Because if I change the color to not transparent, it works. But if I set it transparent I see like the superview, and it's not transparent.

But I repeat, in your example it works, I must be missing something in my code.

**Reply by LeoNatan** [reply]
Can you post a screenshot of Xcode view hierarchy inspector?

**Reply by RiccardoManoni** [reply]
![xcode](https://user-images.githubusercontent.com/17110795/161275441-108ab17e-6151-4051-a5be-bc5206a322c5.png)

In the hierarchy the background seems to be transparent.

**Reply by LeoNatan** [reply]
Check also what’s going on with the _LNPopupBarContentView and _LNPopupBarBackgriundView, and their respective visual effect views.

**Reply by RiccardoManoni** [reply]
![a](https://user-images.githubusercontent.com/17110795/161279080-2d280478-eb54-4da3-8578-adcda325ee0c.png)

It's interesting, the UIView inside _LNPopupBarContentView has the background color I see.
All the other views are correctly transparent. I should try to reach that UIVIew and set it transparent.

**Reply by LeoNatan** [reply]
Also see the alpha of that view. If it is 0, the color doesn’t matter. I think that view is the highlight view, when the user touches the bar.

**Reply by RiccardoManoni** [reply]
I've understand now. There is nothing wrong with the background color. It works, it is transparent.

The problem is that I expected that the collection's viewcells should have been scroll behind the customPopupBar, but this reasoning seems to be wrong.

Do you agree with me?

**Reply by LeoNatan** [reply]
That expectation is not wrong. The content under should go under the popup bar. I’m not sure how you have set up your view hierarchy.
